Parser/tokenizer.c

```diff
@@ -2386,6 +2386,17 @@ tok_get_normal_mode(struct tok_state *tok, tokenizer_mode* current_tok, struct t
               (opening == '[' && c == ']') ||
               (opening == '{' && c == '}')))
         {
+            /* If the opening bracket belongs to an f-string's expression
+               part (e.g. f"{)}") and the closing bracket is an arbitrary
+               nested expression, then instead of matching a different
+               syntactical construct with it; we'll throw an unmatched
+               parentheses error. */
+            if (tok->tok_mode_stack_index > 0 && opening == '{') {
+                int previous_bracket = current_tok->bracket_stack - 1;
```

Comment:
Is possible that we arrive here and `current_tok->bracket_stack` is 0?

Reply:
I don't think so (as long as we consider that `bracket_stack` is incremented for every opening brace when `tok_mode_stack_index > 0`) and this condition is only entered when there is an opening brace matched.

Reply:
Let's add an assert just to be sure

Reply:
good idea, done!